Marketplace/AzureStack.Marketplace.psm1

```diff
@@ -0,0 +1,194 @@
+    }
+
+    Set-AzureRmCurrentStorageAccount -StorageAccountName $storageAccountName -ResourceGroupName $resourceGroupName
+    
+    #same for container
+    $container = Get-AzureStorageContainer -Name $containerName -ErrorAction SilentlyContinue
+    if (-not ($container)) {
+        $container = New-AzureStorageContainer -Name $containerName -Permission Blob
+    }
+
+    if($pscmdlet.ParameterSetName -eq "GalleryItemFromLocal") {
+        $storageAccount.PrimaryEndpoints.Blob
+        $script:galleryItemName = Split-Path $galleryItemLocalPath -Leaf
+        $script:galleryItemBlobURIFromLocal = '{0}{1}/{2}' -f $storageAccount.PrimaryEndpoints.Blob.AbsoluteUri, $containerName,$galleryItemName
+        Set-AzureStorageBlobContent -File $galleryItemLocalPath -Container $containerName -Blob $galleryItemName
+        $galleryItemBlobURI = $galleryItemBlobURIFromLocal
+    }
+
+    if((Get-Module AzureStack).Version -ge [System.Version] "1.2.9"){
+        Add-AzureRMGalleryItem -GalleryItemUri $galleryItemBlobURI
+    }else{
+        Add-AzureRMGalleryItem -SubscriptionId $subscription -GalleryItemUri $galleryItemBlobURI -ApiVersion 2015-04-01
+    }
+
+    Remove-AzureStorageContainer –Name $containerName -Force
+    Remove-AzureRmStorageAccount -ResourceGroupName $resourceGroupName -AccountName $storageAccountName 
+    Remove-AzureRmResourceGroup -Name $resourceGroupName -Force
+}
+
+Function Get-GalleryItem{
+    Param(
+        [ValidateNotNullorEmpty()]
+        [String] $tenantID,
+
+        [Parameter(Mandatory=$true)]
+        [ValidateNotNullorEmpty()]
+        [String] $GalleryItemName
+    )
+
+    $subscription = (Get-AzureStackAdminSubscription -TenantId $tenantId -AzureStackCredentials $azureStackCredentials -ArmEndpoint $ArmEndpoint)
+
+    if((Get-Module AzureStack).Version -ge [System.Version] "1.2.9"){
+        Get-AzureRMGalleryItem -Name $GalleryItemName
+    }else{
+        Get-AzureRMGalleryItem -SubscriptionId $subscription -Name $GalleryItemName -ApiVersion '2015-04-01'
+    }
+}
+
+Function Remove-GalleryItem{
+    Param(
+        [ValidateNotNullorEmpty()]
+        [String] $tenantID,
+
+        [Parameter(Mandatory=$true)]
+        [ValidateNotNullorEmpty()]
+        [String] $GalleryItemName
+    )
+
+    $subscription = (Get-AzureStackAdminSubscription -TenantId $tenantId -AzureStackCredentials $azureStackCredentials -ArmEndpoint $ArmEndpoint)
+
+    if((Get-Module AzureStack).Version -ge [System.Version] "1.2.9"){
+        Remove-AzureRMGalleryItem -Name $GalleryItemName
+    }else{
+        Remove-AzureRMGalleryItem -SubscriptionId $subscription -Name $GalleryItemName -ApiVersion '2015-04-01'
+    }
+}
+
+function Get-AzureStackAdminSubscription {
+    param (
+        [parameter(HelpMessage="Name of the Azure Stack Environment")]
+        [string] $EnvironmentName = "AzureStack",
```

Comment:
Better choose a consistent parameter naming convention, like if you would like to make it all start with lower case then keep it